fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/item/ItemVariant.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.item;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemConvertible;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.network.PacketByteBuf;
+
+import net.fabricmc.fabric.impl.transfer.item.ItemVariantImpl;
+import net.fabricmc.fabric.api.transfer.v1.storage.TransferVariant;
+
+/**
+ * An immutable count-less ItemStack, i.e. an immutable association of an item and an optional NBT compound tag.
+ *
+ * <p>Do not implement, use the static {@code of(...)} functions instead.
+ */
+@ApiStatus.NonExtendable
+public interface ItemVariant extends TransferVariant<Item> {
```

Comment:
override the `hashCode` and `equals` method and state that the method is implemented. I found it annoying in my spigot days when this wasn't documented

Reply:
I'll do it in `TransferVariant` directly.

Reply:
Done.